core/src/test/scala/integration/kafka/server/DynamicBrokerReconfigurationTest.scala

```diff
@@ -216,6 +217,93 @@ class DynamicBrokerReconfigurationTest extends ZooKeeperTestHarness with SaslSet
     verifySslConfig("", invalidSslProperties, configDesc)
   }
 
+  @Test
+  def testUpdatesUsingConfigProvider(): Unit = {
+    val PollingIntervalVal = "${file:polling.interval:interval}"
+    val PollingIntervalUpdateVal = "${file:polling.interval:updinterval}"
+    val SslTruststoreTypeVal = "${file:ssl.truststore.type:storetype}"
+    val SslKeystorePasswordVal = "${file:ssl.keystore.password:password}"
+
+    def executeConfigCommand(props: Properties): Unit = {
+      val propsFile = TestUtils.tempFile()
+      val propsWriter = new FileWriter(propsFile)
+      try {
+        clientProps(SecurityProtocol.SSL).asScala.foreach {
+          case (k, v) => propsWriter.write(s"$k=$v\n")
+        }
+      } finally {
+        propsWriter.close()
+      }
+
+      servers.foreach { server =>
+        val args = buildConfigCommandParameters(server, propsFile.getAbsolutePath, props)
```

Comment:
Hmm. it would be better to create a method `alterConfigsUsingConfigCommand` that does almost everything in this method which was copied from `alterSslKeystoreUsingConfigCommand`.

```
private def alterSslKeystoreUsingConfigCommand(props: Properties, listener: String): Unit = {
    val configPrefix = listenerPrefix(listener)
    val newProps = securityProps(props, KEYSTORE_PROPS, configPrefix)
    alterConfigsUsingConfigCommand(newProps)
   waitForConfig(s"$configPrefix$SSL_KEYSTORE_LOCATION_CONFIG", props.getProperty(SSL_KEYSTORE_LOCATION_CONFIG))
}
```
`executeConfigCommand` can be removed and in its place, you would just call
```
alterConfigsUsingConfigCommand(props)
```

Reply:
Actually it is much better. Thanks for finding this.